components/org.wso2.scim2.testsuite.core/src/main/java/org/wso2/scim2/testsuite/core/tests/BulkTestImpl.java

```diff
@@ -0,0 +1,1643 @@
+        JSONArray jsonarray = jsonObject.optJSONArray("Operations");
+        for (int i = 0; i < jsonarray.length(); i++) {
+            JSONObject innerJsonObject = jsonarray.getJSONObject(i);
+            String location = innerJsonObject.getString("location");
+            userLocations.add(location);
+        }
+        return userLocations;
+    }
+
+    /**
+     * Extract created status from the bulk response.
+     *
+     * @param response Json response from service provider.
+     * @return statusCodes Array of status codes of resources.
+     * @throws JSONException Json exception if response not contain expected attributes.
+     */
+    public ArrayList<Integer> getStatus(String response) throws JSONException {
+
+        ArrayList<Integer> statusCodes = new ArrayList<>();
+        JSONObject jsonObject = new JSONObject(response);
+        JSONArray jsonarray = jsonObject.optJSONArray("Operations");
+        for (int i = 0; i < jsonarray.length(); i++) {
+            JSONObject innerJsonObject = jsonarray.getJSONObject(i);
+            JSONObject innerJsonObject2 = innerJsonObject.getJSONObject("status");
+            Integer code = innerJsonObject2.getInt("code");
+            statusCodes.add(code);
+        }
+        return statusCodes;
+    }
+
+    /**
+     * Create test users for test cases.
+     *
+     * @param noOfUsers Specify the number of users needs to create.
+     * @return userIDS of created users.
+     * @throws ComplianceException        Constructed new exception with the specified detail message.
+     * @throws GeneralComplianceException General exceptions.
+     */
+    private ArrayList<String> createTestsUsers(String noOfUsers) throws ComplianceException,
+            GeneralComplianceException {
+
+        String url = complianceTestMetaDataHolder.getUrl() +
+                ComplianceConstants.TestConstants.USERS_ENDPOINT;
+
+        ArrayList<String> definedUsers = new ArrayList<>();
+        ArrayList<String> userIDs = new ArrayList<>();
+
+        if (noOfUsers.equals("One")) {
+            definedUsers.add(ComplianceConstants.DefinedInstances.defineUser);
+        } else if (noOfUsers.equals("Many")) {
+            definedUsers.add(ComplianceConstants.DefinedInstances.definedUser1);
+            definedUsers.add(ComplianceConstants.DefinedInstances.definedUser2);
+            definedUsers.add(ComplianceConstants.DefinedInstances.definedUser3);
+            definedUsers.add(ComplianceConstants.DefinedInstances.definedUser4);
+            definedUsers.add(ComplianceConstants.DefinedInstances.definedUser5);
+        }
+
+        HttpPost method = new HttpPost(url);
+        // Create users.
+        HttpClient client = HTTPClient.getHttpClient();
+        method = (HttpPost) HTTPClient.setAuthorizationHeader(complianceTestMetaDataHolder, method);
+        method.setHeader("Accept", "application/json");
+        method.setHeader("Content-Type", "application/json");
```

Comment:
Words such as `Accept`, `Content-Type`, `application/json` are reused in many classes. Define them in a common constant class.
Practice the same for other places